### ISSUE TITLE
Investigate and address -Wparentheses-equality warnings in compiler

### ISSUE THREAD

**Issue by 0xdaryl** [eclipse/omr#1648]
Investigate and fix:
```
warning: equality comparison with extraneous parentheses [-Wparentheses-equality]
```
(e.g. ../compiler/optimizer/RedundantAsyncCheckRemoval.cpp:833)

**Comment by charliegracie** [comment]
I believe these were all resolved via #1635

**Comment by 0xdaryl** [comment]
These do appear to be resolved now.  Closing...
